components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/rich_search_widget.tsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { loadTimeData } from '$web-common/loadTimeData'
+import * as React from 'react'
+import styles from './rich_search_widget.module.scss'
+
+const RICH_SEARCH_WIDGETS_ORIGIN = loadTimeData.getString(
+  'richSearchWidgetsOrigin',
+)
+export default function RichSearchWidget(props: { jsonData: string }) {
+  const frameHandler = React.useCallback(
+    (iframe: HTMLIFrameElement | null) => {
+      if (iframe) {
+        const sendContent = () => {
+          iframe.contentWindow?.postMessage(
+            JSON.parse(props.jsonData),
+            RICH_SEARCH_WIDGETS_ORIGIN,
+          )
+        }
+
+        // Send content immediately (in case its already loaded) and when the iframe loads.
+        sendContent()
+        iframe.onload = sendContent
+
+        window.addEventListener('message', (event) => {
+          // Ignore messages from other origins.
+          if (event.origin !== RICH_SEARCH_WIDGETS_ORIGIN) {
+            return
+          }
+
+          if (
+            event.data.type !== 'resize'
+            || event.source !== iframe.contentWindow
+          ) {
+            return
+          }
+          iframe.style.height = event.data.height + 'px'
+        })
+      }
+    },
+    [props.jsonData],
+  )
+
+  return (
+    <iframe
+      className={styles.richSearchWidget}
+      sandbox='allow-scripts allow-same-origin'
```

Comment:
@diracdeltas mind taking a look at this `sandbox` attribute? I think it should be fine because its being served from a different origin.